### PR TITLE
Second HomeWork completed

### PR DESCRIPTION
**Pull request by NikolayFatkullin**
- Created annotation @Dao, @Inject;
- Created classes Factory, Injector, AnnotationException;

### CODE REVIEW

**Comment by boroda4436** [comment]
I haven't reviewed your code. Just launched. This is not a rejection reason, just advice. Your app will be better if the user will not see exceptions like `IndexOutOfBounds` or `NullPointerException`.
image

**Review by Fedich09 — APPROVED** [review]
good joberino))

File: src/main/java/core/basesyntax/lib/Injector.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+public class Injector {
+    public static Object getInstance(Class<?> clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        Constructor<?> constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/lib/Injector.java:19]: FYI you use here isAnnotationPresent as well (it's optional)

File: src/main/java/core/basesyntax/lib/Injector.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                field.setAccessible(true);
+                Object dao = new Object();
+                if (field.getType() == BetDao.class) {
+                    dao = Factory.getBetDao();
+                } else if (field.getType() == PlayerDao.class) {
+                    dao = Factory.getPlayerDao();
+                }
+                if (dao.getClass().getAnnotation(Dao.class) == null) {
+                    throw new AnnotationException("This type of class doesn't have "
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/lib/Injector.java:28]: add type to message to make more informative

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]
good job, left one minor comment, rest is good

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**